### DISCUSSION TITLE
Making Weston/Wayland functional on Jetson TX2

### DISCUSSION THREAD

**Discussion by TorEldby** [OE4T/tegra-demo-distro#156] [Q&A]
Hello
I have tried to follow along the implementation of Weston, but have hit a roadblock.
There was no issue with building from the repo itself, but when I tried to implement it myself with a distro compatible with tegra and mender, I believe I have missed some parts.
In essence, I simply copied the packagegroups and included them in the image using IMAGE_INSTALL, similar to the original CORE_IMAGE_BASE_INSTALL. Otherwise I do not really see much I can do on my end. It builds fine and installs all the correct packages, but it does not launch with a functional GUI manager or anything, and simply launches into tty. Running "echo $XDG_SESSION_TYPE" confirms this, as it outputs "tty", rather than "weston".
Granted, I am pretty new to Yocto, so I believe there is some .bbappend file or otherwise configuration I have missed which I need to do as well to make weston run by default, as with what is done with this repo's configuration.

My question is: Is there some configuration I am missing aside from the package installations, of which I haven't found by studying the repository, which I need to make Weston successfully run on startup?
Thank you.

**Comment by dwalkes** [comment]
>  but when I tried to implement it myself with a distro compatible with tegra and mender, I believe I have missed some parts.

Did you see/follow the instructions in https://github.com/OE4T/tegra-demo-distro/wiki/Creating-your-Distro-Layer ?  Which branch did you use?
